crates/wasm-mutate-stats/src/bin/wasm-mutate-stats.rs

```diff
@@ -0,0 +1,327 @@
+                                wasm = mutated;
+                            }
+                            Err(_) => { /* bypass */ }
+                        }
+                    }
+                    println!("Reached");
+                    while report.rotations <= ROTATIONS {
+                        report.rotations += 1;
+                        match wasmmutate.run_with_rng(&wasm, &mut rng) {
+                            Ok(mutated) => {
+                                let mut validator = wasmparser::Validator::new();
+                                match validator.validate_all(&mutated) {
+                                    Ok(_) => {
+                                        match self.compile(&mutated, &name, report.rotations) {
+                                            Ok(mhashes) => {
+                                                let hash = self.hash(mutated.clone());
+                                                wasm = mutated;
+                                                report.add_wasm_hash(hash);
+
+                                                for (idx, mhash) in mhashes.iter().enumerate() {
+                                                    report.add_obj_hash(mhash.clone(), idx);
+                                                }
+
+                                                if report.rotations % 100 == 0 {
+                                                    // Save report
+                                                    self.save_report(&report, &name);
+                                                    if self.print {
+                                                        report.print();
+                                                    }
+                                                }
+                                            }
+                                            Err(e) => report
+                                                .errors
+                                                .push((report.rotations as u64, format!("{}", e))),
+                                        }
+                                    }
+                                    // All generated Wasm should be valid
+                                    Err(e) => panic!("{}", e),
+                                }
+                            }
+                            Err(e) => report
+                                .errors
+                                .push((report.rotations as u64, format!("{}", e))),
+                        }
+                    }
+                }
+                None => {
+                    break;
+                }
+            }
+        }
+    }
+
+    fn hash(&self, data: Vec<u8>) -> String {
+        // By default md5 is the hashes here
+        format!("{:x}", Md5::digest(&data))
+    }
+
+    /// Compile a module using `mutated` as the input Wasm
+    ///
+    fn compile(
+        &self,
+        mutated: &Vec<u8>,
```

Comment:
Passing `&Vec<u8>` is basically the same as passing `&[u8]` but doesn't allow callers who have a `&[u8]` to call the function, forcing them to allocate a new vec and pass a reference to it unnecessarily.

```suggestion
        mutated: &[u8],
```